api/src/pcapi/scripts/reindex_offers_without_none_fields/main.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+import decimal
+import logging
+import statistics
+import time
+
+import pytz
+
+from pcapi.core import search
+import pcapi.core.offers.models as offers_models
+from pcapi.core.search.backends.algolia import AlgoliaBackend
+from pcapi.core.search.backends.algolia import REDIS_HASHMAP_INDEXED_OFFERS_NAME
+
+
+logger = logging.getLogger(__name__)
+
+BATCH_SIZE = 1_000
+REPORT_EVERY = 10_000
+
+
+def serialize_offer_without_none_fields(cls, offer: offers_models.Offer, last_30_days_bookings: int) -> dict:  # type: ignore[no-untyped-def]
+    result = AlgoliaBackend._original_serialize_offer(offer, last_30_days_bookings)  # type: ignore[attr-defined]
+    for section in ("offer", "offerer", "venue"):
+        result[section] = {key: value for key, value in result[section].items() if value is not None}
+    return result
+
+
+AlgoliaBackend._original_serialize_offer = AlgoliaBackend.serialize_offer  # type: ignore[attr-defined]
+AlgoliaBackend.serialize_offer = serialize_offer_without_none_fields  # type: ignore[method-assign, assignment]
+
+
+def _get_eta(end: int, current: int, elapsed_per_batch: list[int]) -> str:
+    left_to_do = end - current
+    eta_seconds = left_to_do / BATCH_SIZE * statistics.mean(elapsed_per_batch)
+    eta_datetime = datetime.datetime.utcnow() + datetime.timedelta(seconds=eta_seconds)
+    eta_datetime = eta_datetime.astimezone(pytz.timezone("Europe/Paris"))
+    return eta_datetime.strftime("%d/%m/%Y %H:%M:%S")
+
+
+def full_reindex_offers() -> None:
+    backend = AlgoliaBackend()
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🤓 nitpick:** tu aurais pu faire un `get_backend` ici au cas où le script des lancé par erreur en testing ou développement‏

Reply:
Le script peut tourner en testing sans soucis, et en development il ne devrait pas y avoir les secrets pour utiliser l'API d'Algolia.
Vu qu'on utilise la hashmap de redis pour savoir quelles offres sont indexées, qui n'est utilisée que par `AlgoliaBackend`, le script avec un autre backend n'a pas de sens.
Sinon je suis d'accord avec toi, bien vu, mais dans ce cas particulier je vais laisser comme ça :)